webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/CheckNcfTechniqueUpdate.scala

```diff
@@ -90,104 +80,48 @@ class CheckNcfTechniqueUpdate(
-                                case Full(techniques) => Right(techniques)
+                                  match {
+                                  case Full(m) => m.succeed
+                                  case eb : EmptyBox =>
+                                    val fail = eb ?~! "An error occured while writing ncf Techniques"
+                                    WriteTechniqueError(fail.messageChain, None).fail
+                                }
+                              )
+                                                        // Update technique library once all technique are updated
+        libUpdate   <- techLibUpdate.update(ModificationId(uuidGen.newUuid), EventActor(systemApiToken.name.value), Some(s"Update Technique library after updating all techniques at start up")) match {
+                                case Full(techniques) => techniques.succeed
                                 case eb: EmptyBox =>
                                   val fail = eb ?~! s"An error occured during techniques update after update of all techniques from the editor"
-                                  Left(TechniqueUpdateError(fail.msg, fail.exception))
+                                  TechniqueUpdateError(fail.msg, fail.exception).fail
                               }
-        flagDeleted        <- tryo (
-                                  ncfTechniqueUpdateFlag.delete()
-                                , s"An error occured while deleting file ${ncfTechniqueUpdateFlag} after techniques were written, you may need to delete it manually"
-                                , FlagFileError
-                              )
+        flagDeleted <- IOResult.effect( ncfTechniqueUpdateFlag.delete() )
       } yield {
          techniques
-      } ) match {
-        case Right(techniques) =>
-          BootstrapLogger.logEffect.info(s"All ncf techniques were updated, ${techniques.size} were migrated")
-        case Left(NcfApiAuthFailed(msg,e)) =>
-          BootstrapLogger.logEffect.warn(s"Could not authenticate in ncf API, maybe it was not initialized yet, retrying in 5 seconds, details ${msg}")
-          ZioRuntime.internal.unsafeRunAsync(Task.effect(updateNcfTechniques).delay(5.seconds))( _ => ())
-        case Left(FlagFileError(_,_)) =>
-          BootstrapLogger.logEffect.warn(s"All ncf techniques were updated, but we could not delete flag file ${ncfTechniqueUpdateFlag}, please delete it manually")
-        case Left(e : NcfTechniqueUpgradeError) =>
-          BootstrapLogger.logEffect.error(s"An error occured while updating ncf techniques: ${e.msg}")
       }
     }
 
     try {
       if (ncfTechniqueUpdateFlag.exists) {
-        ZioRuntime.internal.unsafeRunAsync(Task.effect(updateNcfTechniques).delay(10.seconds))( _ => ())
+        ZioRuntime.runNow(updateNcfTechniques)
```

Comment:
You need to handle errors here to avoid stack traces:
```
val prog = updateNcfTechniques.catchAll(err => theCorrectLogger.error(err.chainError("Error when updating ncf techniques").fullMsg)